plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go

```diff
@@ -0,0 +1,163 @@
+	Profile         string `json:"profile"          jsonschema:"title=profile,description=The Profile to use to create the session, env var AWS_PROFILE if present"`
+	Region          string `json:"region"           jsonschema:"title=region,description=The Region of your EKS cluster, env var AWS_REGION is used if present"`
+	BufferSize      uint64 `json:"buffer_size"      jsonschema:"title=buffer_size,description=Buffer Size (default: 200),default=200"`
+	Shift           uint64 `json:"shift"            jsonschema:"title=shift,description=Time shift in past in seconds (default: 1s),default=1"`
+	PollingInterval uint64 `json:"polling_interval" jsonschema:"title=polling_interval,description=Polling Interval in seconds (default: 5s),default=5"`
+	UseAsync        bool   `json:"use_async"        jsonschema:"title=use_async,description=If true then async extraction optimization is enabled (default: true),default=true"`
+}
+
+func (k *Plugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          9,
+		Name:        pluginName,
+		Description: "Read Kubernetes Audit Events for EKS from Cloudwatch Logs",
+		Contact:     "github.com/falcosecurity/plugins",
+		Version:     "0.1.0",
+		EventSource: "k8s_audit",
+	}
+}
+
+// Reset sets the configuration to its default values
+func (p *PluginConfig) Reset() {
+	if i := os.Getenv("AWS_DEFAULT_PROFILE"); i != "" {
+		p.Profile = i
+	}
+	if i := os.Getenv("AWS_PROFILE"); i != "" {
+		p.Profile = i
+	}
+	if i := os.Getenv("AWS_DEFAULT_REGION"); i != "" {
+		p.Region = i
+	}
+	if i := os.Getenv("AWS_REGION"); i != "" {
+		p.Region = i
+	}
+	p.UseAsync = true
+	// for PollingInterval, Shift and BufferSize, the default values from the package are used automatically
+}
+
+func (k *Plugin) Init(cfg string) error {
+	// read configuration
+	k.Plugin.Config.Reset()
+	k.Config.Reset()
+
+	err := json.Unmarshal([]byte(cfg), &k.Config)
+	if err != nil {
+		return err
+	}
+
+	// setup optional async extraction optimization
+	extract.SetAsync(k.Config.UseAsync)
+
+	k.Logger = log.New(os.Stderr, "["+pluginName+"] ", log.LstdFlags|log.LUTC|log.Lmsgprefix)
+
+	return nil
+}
+
+func (p *Plugin) InitSchema() *sdk.SchemaInfo {
+	reflector := jsonschema.Reflector{
+		// all properties are optional by default
+		RequiredFromJSONSchemaTags: true,
+		// unrecognized properties don't cause a parsing failures
+		AllowAdditionalProperties: true,
+	}
+	if schema, err := reflector.Reflect(&PluginConfig{}).MarshalJSON(); err == nil {
+		return &sdk.SchemaInfo{
+			Schema: string(schema),
+		}
+	}
+	return nil
+}
+
+func (p *Plugin) OpenParams() ([]sdk.OpenParam, error) {
+	return []sdk.OpenParam{
+		{Value: "default", Desc: "Cluster Name"},
+	}, nil
+}
+
+func (p *Plugin) Open(clustername string) (source.Instance, error) {
+	if clustername == "" {
+		return nil, fmt.Errorf("cluster name can't be empty")
+	}
+	filter := cloudwatchlogs.CreateFilter("", "/aws/eks/"+clustername+"/cluster", "kube-apiserver-audit", nil)
```

Comment:
aws supports two filter per log group which is not many, it's probably a good idea to support using the existing filter as an option.

Reply:
I don't understand your point, the idea here is to create a nice UX, the users just set the name of their cluster and the filter is generated by the plugin.

Reply:
Ok, I didn't used a subscription, so we're not limited in number of filters per log group. Should be ok.